### PR TITLE
Homework ten

### PR DESCRIPTION
**Pull request by ibarrae**
Rearranged newtypes for better readability

Finished exercise one of homework ten

Finished exercise two of homework ten

Finished exercise three from homework ten

Finished exercise five of homework ten

### CODE REVIEW

File: src/HomeworkTen/AParser.hs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+module HomeworkTen.AParser where
+
+import Data.Char
+import Control.Applicative
+
+-- A parser for a value of type a is a function which takes a String
+-- represnting the input to be parsed, and succeeds or fails; if it
+-- succeeds, it returns the parsed value along with the remainder of
+-- the input.
+newtype Parser a
```
Comment (sestrella) [inline src/HomeworkTen/AParser.hs]: @kinorius123 could you please write some property test to check that our instances of Functor and Applicative fulfill all the laws.

Comment (kinorius123) [reply src/HomeworkTen/AParser.hs]: @sestrella, I just added the property test for the parser newtype in a new commit.

File: src/HomeworkTen/AParser.hs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+intPair :: Parser [Integer]
+intPair = 
+    let f x _ z = [x,z]
+    in liftA3 f posInt (char ' ') posInt
+
+intOrUppercase :: Parser ()
+intOrUppercase =
+    let f _ = ()
+        int = f <$> posInt
+        upper = f <$> satisfy isUpper
```
Comment (dbalseiro) [inline src/HomeworkTen/AParser.hs]: Take a look at `const`
http://hackage.haskell.org/package/base-4.11.1.0/docs/Prelude.html#v:const

File: test/AParserSpec.hs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import HomeworkTen.AParser
+
+spec :: Spec
+spec = do
+    describe "first" $ do
+        it "returns a tuple" $
+            first (+1) ((1,2) :: (Int, Int)) `shouldBe` (2,2)
+        it "returns a tuple" $
+            first (>0) ((1,2) :: (Int, Int)) `shouldBe` (True,2)
+    describe "abParser" $ do
```
Comment (dbalseiro) [inline test/AParserSpec.hs]: I think you are missing contexts in your tests... take a look at http://www.betterspecs.org/

**Pull request merged**